### PR TITLE
api: add sender support

### PR DESCRIPTION
**Pull request by dcow**
Needs more work so we test more than `None`, but should be enough
for now to integrate with.

### CODE REVIEW

File: api/src/mailbox.rs [hunk 1]
```diff
@@ -37,18 +37,28 @@ pub struct MessageToDelete
...
     pub uuid: String,
     pub data: Payload,
 }
 
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+pub struct Sender
+{
+    pub signing_key: String,
+    pub encryption_key: String,
+    pub cid: String,
```
Comment (tremblay) [inline api/src/mailbox.rs:51]: do these need to be marked as optional? I'm not familiar with rust schemas

Comment (dcow) [reply api/src/mailbox.rs:51]: Yeah sry

**Review by tremblay — APPROVED** [review]
